cmake/modules/Findcrypto.cmake

```diff
@@ -56,7 +56,7 @@ else()
     )
 
     if (NOT crypto_LIBRARY)
-        if (BUILD_SHARED_LIBS)
+        if (BUILD_SHARED_LIBS OR ${S2N_USE_CRYPTO_SHARED_LIBS})
```

Comment:
It seems like cmake doesn't like the syntax here:
```
CMake Error at /tmp/tmp.xcMT5KADjW/s2n-install-shared/lib/s2n/cmake/modules/Findcrypto.cmake:59 (if):
  if given arguments:

    "BUILD_SHARED_LIBS" "OR"

  Unknown arguments specified
```
Would something like this maybe work?
```suggestion
        if (${BUILD_SHARED_LIBS} OR ${S2N_USE_CRYPTO_SHARED_LIBS})
```

Reply:
what cmake version are you using? its working as expected on my machine? got a image i can use to reproduce this?

Reply:
this works on a fresh al2023 image

```
FROM public.ecr.aws/amazonlinux/amazonlinux:2023

# Install dev tools
RUN yum groupinstall -y "Development Tools"

# Install cmake and ninja
RUN yum install -y cmake3 ninja-build

# Install static and dynamic lc
RUN git clone --depth 1 -b fips-2022-11-02 https://github.com/aws/aws-lc && \
    cd aws-lc && \
    mkdir build && \
    cd build && \
    cmake -G Ninja -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_INSTALL_PREFIX=/lc-install .. && \
    cmake --build . && \
    cmake --install . && \
    rm -rf ./* && \
    cmake -G Ninja -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_INSTALL_PREFIX=/lc-install .. && \
    cmake --build . && \
    cmake --install .

RUN git clone https://github.com/sbiscigl/s2n-tls.git # && \
    cd s2n-tls && \
    mkdir build && \
    cd build && \
    cmake -G Ninja -DCMAKE_PREFIX_PATH=/lc-install .. && \
    cmake --build .
```